pkg/athena/driver/rows.go

```diff
@@ -41,27 +40,27 @@ func (r *Rows) ColumnTypeScanType(index int) reflect.Type {
 
 func (r *Rows) athenaTypeOf(columnInfo *athena.ColumnInfo) (reflect.Type, error) {
 	switch *columnInfo.Type {
-	case "tinyint":
-		return reflect.TypeOf(int8(0)), nil
-	case "smallint":
-		return reflect.TypeOf(int16(0)), nil
+	case "tinyint", "smallint":
+		return reflect.TypeOf(sql.NullInt16{}), nil
 	case "integer":
-		return reflect.TypeOf(int32(0)), nil
+		return reflect.TypeOf(sql.NullInt32{}), nil
 	case "bigint":
-		return reflect.TypeOf(int64(0)), nil
-	case "float", "real":
-		return reflect.TypeOf(float32(0)), nil
-	case "double":
-		return reflect.TypeOf(float64(0)), nil
+		return reflect.TypeOf(sql.NullInt64{}), nil
```

Comment:
I was wondering about this after something @fridgepoet and I discussed this morning: In [this PR](https://github.com/grafana/opensearch-datasource/pull/168/files#r1200080425) she asserted nullable values as &f instead of using an external library. Would this be possible here? Something like: 
```reflect.TypeOf(&int64(0)), nil```

Reply:
I get an error when I try to use `invalid operation: cannot take address of int64(0) (constant 0 of type int64)compilerUnaddressableOperand`

Reply:
You would have to do something like this:
```golang
case "tinyint", "smallint":
	someVarName := int8(0)
	return reflect.TypeOf(&someVarName), nil
```

Reply:
there's an error `sql: Scan error on column index 0, name \"_col0\": converting NULL to int8 is unsupported: Could not process SQL results`. the `sql.Null*` types implement the `Scanner` interface which allows handling possibly null values https://pkg.go.dev/database/sql#NullInt64